lib/src/Scene/Qt/QtItemTools.cpp

```diff
@@ -72,28 +107,104 @@ QObject* FindChildItem(QObject* object, const QString& name)
     if (auto qquickitem = qobject_cast<const QQuickItem*>(object)) {
         for (Index i = 0; i < qquickitem->childItems().size(); ++i) {
             auto child = qquickitem->childItems().at(i);
+            if (name.has_value()) {
+                if (GetObjectName(child) == name) {
+                    return child;
+                }
+
+                if (auto item = FindChildItem(child, name, {}, {}, {})) {
+                    return item;
+                }
+            }
+
             if (GetObjectName(child) == name) {
                 return child;
             }
-            if (auto item = FindChildItem(child, name)) {
-                return item;
+            if (propertyName.has_value() && propertyValue.has_value()) {
+                if (PropertValueByObject(child, propertyName.value()) == propertyValue.value()) {
+                    return child;
+                }
+
+                if (auto item = FindChildItem(child, name, propertyName.value(), propertyValue.value(), {})) {
+                    return item;
+                }
+            } else if (type.has_value()) {
+                if (TypeByObject(child) == type.value()) {
+                    return child;
+                }
+
+                if (auto item = FindChildItem(child, name, {}, {}, type)) {
+                    return item;
+                }
+
+            } else {
+                if (auto item = FindChildItem(child, name)) {
+                    return item;
+                }
             }
         }
     } else {
         for (Index i = 0; i < object->children().size(); ++i) {
             auto child = object->children().at(i);
-            if (GetObjectName(child) == name) {
+            if (name.has_value() && GetObjectName(child) == name) {
                 return child;
             }
-            if (auto item = FindChildItem(child, name)) {
-                return item;
+            if (propertyValue.has_value() && propertyName.has_value()) {
+                if (auto item = FindChildItem(child, name, propertyName.value(), propertyValue.value(), {})) {
+                    return item;
+                }
+            } else if (type.has_value()) {
+                if (auto item = FindChildItem(child, name, {}, {}, type)) {
+                    return item;
+                }
+            } else {
+                if (auto item = FindChildItem(child, name)) {
+                    return item;
+                }
             }
```

Comment:
I think this could just be a single
```suggestion
            if (auto item = FindChildItem(child, name, propertyName, propertyValue, type)) {
                return item;
            }
```

Where the optionals are just passed along, leaving the `has_value()` check to the `FindChildItem` call.